pkg/resizer/csi_resizer_test.go

```diff
@@ -66,8 +71,218 @@ func TestNewResizer(t *testing.T) {
 	}
 }
 
-func fakeK8s() (kubernetes.Interface, informers.SharedInformerFactory) {
-	client := fake.NewSimpleClientset()
+func TestResizeMigratedPV(t *testing.T) {
+	testCases := []struct {
+		name               string
+		driverName         string
+		pv                 *v1.PersistentVolume
+		nodeResizeRequired bool
+		err                error
+	}{
+		{
+			name:               "Test AWS EBS CSI Driver",
+			driverName:         "ebs.csi.aws.com",
+			pv:                 createEBSPV(1),
+			nodeResizeRequired: true,
+		},
+		{
+			name:               "Test GCE PD Driver",
+			driverName:         "pd.csi.storage.gke.io",
+			pv:                 createGCEPDPV(1),
+			nodeResizeRequired: true,
+		},
+		{
+			name:               "Test unknonwn driver",
+			driverName:         "unknown",
+			pv:                 createEBSPV(1),
+			nodeResizeRequired: true,
+			err:                errors.New("volume testEBSPV is not migrated to CSI"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			driverName := tc.driverName
+			client := csi.NewMockClient(driverName, true, true, true)
+			k8sClient, informerFactory := fakeK8s()
+			resizer, err := NewResizerFromClient(client, 0, k8sClient, informerFactory)
+			if err != nil {
+				t.Fatalf("Failed to create resizer: %v", err)
+			}
+
+			pv := tc.pv
+			expectedSize := quantityGB(2)
+			newSize, nodeResizeRequired, err := resizer.Resize(pv, expectedSize)
+
+			if tc.err != nil {
+				if err == nil {
+					t.Fatalf("Got wrong error, wanted: %v, got: %v", tc.err, err)
+				}
+			} else {
+				if err != nil {
+					t.Fatalf("Failed to resize the PV: %v", err)
+				}
+
+				if newSize != expectedSize {
+					t.Fatalf("newSize mismatches, wanted: %v, got: %v", expectedSize, newSize)
+				}
+				if nodeResizeRequired != tc.nodeResizeRequired {
+					t.Fatalf("nodeResizeRequired mismatches, wanted: %v, got: %v", tc.nodeResizeRequired, nodeResizeRequired)
+				}
+			}
+		})
+	}
+}
+
+func TestCanSupport(t *testing.T) {
+	testCases := []struct {
+		name       string
+		driverName string
+		pv         *v1.PersistentVolume
+		pvc        *v1.PersistentVolumeClaim
+		canSupport bool
+	}{
+		{
+			name:       "EBS PV/PVC is supported",
+			driverName: "ebs.csi.aws.com",
+			pv:         createEBSPV(1),
+			pvc:        createPVC(2, 1, "ebs.csi.aws.com"),
+			canSupport: true,
+		},
+		{
+			name:       "EBS PV/PVC is not supported when migartion is disabled",
+			driverName: "ebs.csi.aws.com",
+			pv:         createEBSPV(1),
+			pvc:        createPVC(2, 1, "kubernetes.io/aws-ebs"),
```

Comment:
its not clear to me why setting the pvc annotation to this means "migration is disabled", this could probably be fixed by more comments on the annotation key that I mentioned in the other comment

Reply:
yeah. It means, when CSI driver is installed, but migration is not enabled. Updated the comment for the key